usl_models/usl_models/flood_ml/dataset.py

```diff
@@ -16,6 +16,91 @@
+    firestore_client: Optional[firestore.Client] = None,
+    storage_client: Optional[storage.Client] = None,
+) -> tf.data.Dataset:
+    """Creates a dataset which generates chunks for flood model inference.
+
+    For training with teacher-forcing, `load_dataset_windowed` should be used instead.
+    The examples are generated from multiple simulations.
+    The dataset iteratively yields examples read from Google Cloud Storage to avoid
+    pulling all examples into memory at once.
+
+    Args:
+      sim_names: The simulation labels to use for training,
+                 e.g. ["Manhattan-config_v1/Rainfall_Data_1.txt"]
+      batch_size: Size of batches yielded by the dataset. Approximate memory
+                  usage is 10GB * batch_size during training.
+      n_flood_maps: The number of flood maps in each example.
+      m_rainfall: The width of the temporal rainfall tensor.
+      max_chunks: The maximum number of examples to yield from the dataset.
+                  If `None` (default) yields all examples from the simulations.
+      firestore_client: The client to use when interacting with Firestore.
+      storage_client: The client to use when interacting with Cloud Storage.
+    """
+    firestore_client = firestore_client or firestore.Client()
+    storage_client = storage_client or storage.Client()
+
+    def generator():
+        """Generator for producing full inputs and labels."""
+        for sim_name in sim_names:
+            for model_input, labels in _iter_model_inputs(
+                firestore_client,
+                storage_client,
+                sim_name,
+                n_flood_maps,
+                m_rainfall,
+                max_chunks,
+            ):
+                yield model_input, labels
+
+    # Create the dataset for this simulation
+    dataset = tf.data.Dataset.from_generator(
+        generator=generator,
+        output_signature=(
+            dict(
+                geospatial=tf.TensorSpec(
+                    shape=(
+                        constants.MAP_HEIGHT,
+                        constants.MAP_WIDTH,
+                        constants.GEO_FEATURES,
+                    ),
+                    dtype=tf.float32,
+                ),
+                temporal=tf.TensorSpec(
+                    shape=(constants.MAX_RAINFALL_DURATION, m_rainfall),
+                    dtype=tf.float32,
+                ),
+                spatiotemporal=tf.TensorSpec(
+                    shape=(
+                        n_flood_maps,
+                        constants.MAP_HEIGHT,
+                        constants.MAP_WIDTH,
+                        1,
+                    ),
+                    dtype=tf.float32,
+                ),
+            ),
+            tf.TensorSpec(
+                shape=(None, constants.MAP_HEIGHT, constants.MAP_WIDTH),
+                dtype=tf.float32,
+            ),
+        ),
+    )
+    # If no batch specified, do not batch the dataset, which is required
```

Comment:
Is the batch_size `None` in that case? Or is it 0?
If `None` it'd be good to do
```python
batch_size: Optional[int] = 4,
```
above.
And also have 
```python
if batch_size is not None
```
here

Reply:
We've been using batch_size = 0.